services/auth.js

```diff
@@ -31,8 +36,20 @@ const authService = {
 
     const hashedPassword = await hashPassword(password)
     const activationLink = uuidv4()
+    const foundRole = await Role.findOne({ value: role }).exec()
+
+    if (!foundRole) {
+      throw createError(404, ROLE_NOT_SUPPORTED)
+    }
 
-    const user = await User.create({ role, firstName, lastName, email, password: hashedPassword, activationLink })
+    const user = await User.create({
+      role: foundRole,
```

Comment:
Here, from what I see, you're saving the whole object role instead of just saving the role id, while when creating the super admin (code above), you actually _did_ save just the id.

Reply:
nice catch, but since i set Schema.Types.ObjectId as a type in User model for the role field it takes and saves id only to the DB from the Role collection, not the whole object.